src/libcrun/utils.c

```diff
@@ -530,7 +530,7 @@ crun_safe_ensure_at (bool do_open, bool dir, int dirfd, const char *dirpath,
       if (it == NULL)
         break;
 
-      cur = it + 1;
+      cur = consume_slashes (it + 1);
```

Comment:
Why not realpath?

Reply:
`realpath` doesn't deal with the race conditions we deal with in this function, and it would be relatively too expensive to polish the path element.